### PR TITLE
[dvsim] Allow non-integral values of --reseed-multiplier

### PR DESCRIPTION
**Pull request by rswarbrick**

### CODE REVIEW

File: util/dvsim/SimCfg.py [hunk 1]
```diff
@@ -392,8 +392,11 @@ def prune_items(items, marked_items):
             if self.reseed_ovrd is not None:
                 test.reseed = self.reseed_ovrd
 
-            # Apply reseed multiplier if set on the command line.
-            test.reseed *= self.reseed_multiplier
+            # Apply reseed multiplier if set on the command line. This is
+            # always positive but might not be an integer. Round to nearest,
+            # but pick at least one.
+            scaled = int(test.reseed * self.reseed_multiplier + 0.5)
```
Comment (imphil) [inline util/dvsim/SimCfg.py:398]: nit: I'd go for `math.ceil()`.

File: util/dvsim/SimCfg.py [hunk 2]
```diff
@@ -392,8 +392,11 @@ def prune_items(items, marked_items):
             if self.reseed_ovrd is not None:
                 test.reseed = self.reseed_ovrd
 
-            # Apply reseed multiplier if set on the command line.
-            test.reseed *= self.reseed_multiplier
+            # Apply reseed multiplier if set on the command line. This is
+            # always positive but might not be an integer. Round to nearest,
+            # but pick at least one.
```
Comment (imphil) [inline util/dvsim/SimCfg.py:397]:
```suggestion
            # but pick at least 1.
```

(I read it initially as "pick at least one multiplier" or something like that.)

**Review by imphil — APPROVED** [review]

File: util/dvsim/SimCfg.py (hunk 1 above)
Comment (rswarbrick) [reply util/dvsim/SimCfg.py]: Good point, although this is supposed to be `round()`: a strong argument for using a named function!

File: util/dvsim/SimCfg.py (hunk 2 above)
Comment (rswarbrick) [reply util/dvsim/SimCfg.py]: Oh! That was super-unclear! Fixed.

**Pull request merged**